mlir-sycl/doc/design/SYCLGridRepresentation.md

```diff
@@ -98,110 +91,25 @@ file also generates call to register it with no further action.
 
 In the codegen side, right before generating a `sycl.call` operation, we check
 whether there is an operation implementing `SYCLMethodOpInterface` homologous to
-that function call. In order to do that, `SYCLDialect::findMethod(mlir::TypeID,
-llvm::StringRef)` can be used, passing the ID of the SYCL type the function is a
-member of and the function name. After looking up the name in its registry,
-`llvm::None` is returned if no operation can be used to replace the `sycl.call`;
-otherwise, the name of the operation is returned, which can be used to create a
-different operation instead.
+that function call. In order to do that,
+`SYCLDialect::lookupMethod(mlir::TypeID, llvm::StringRef)` can be used, passing
+the ID of the SYCL type the function is a member of and the function name. After
+looking up the name in its registry, `llvm::None` is returned if no operation
+can be used to replace the `sycl.call`; otherwise, the name of the operation is
+returned, which can be used to create a different operation instead.
 
 Note that, in case the function is a member of a base type, the `sycl.cast`
 operation used to cast to the base class is abstracted beforehand.
 
-### Canonicalization
-
-To help the analysis, canonicalization rules are introduced in such a way that
-the total number of operations is minimized, i.e., we reduce the number of
-operations by replacing combinations of operations by equivalent
-single-operations, e.g., the following code:
-
-```MLIR
-%0 = sycl.nd_item.get_global_id(%nd) {FunctionArgTypes = {!llvm.ptr<struct<...>, 4>}, FunctionName = @get_global_id, TypeName = @nd_item} : (!sycl_nd_item_1_) -> !sycl_id_1_
-%1 = sycl.id.get(%0, %idx) {FunctionArgTypes = {memref<?x!sycl_id_1_, 4>, i32}, FunctionName = @get, TypeName = @id} : (!sycl_id_1_, i32) -> i64
-```
-
-would be simplified to:
-
-```MLIR
-%1 = sycl.nd_item.get_global_id(%nd, %idx) {FunctionArgTypes = {!llvm.ptr<struct<...>, 4>, i32}, FunctionName = @get_global_id, TypeName = @nd_item} : (!sycl_nd_item_1_, i32) -> i64
-```
```

Comment:
We still need to implement some canonicalization rules, but this will not be like this as we use reference semantics now